test-clients/src/main/java/com/hedera/services/bdd/suites/contract/precompile/PauseUnpauseTokenAccountPrecompileSuite.java

```diff
@@ -0,0 +1,350 @@
+                                                                PAUSE_UNPAUSE_CONTRACT,
+                                                                UNPAUSE_TOKEN_ACCOUNT_FUNCTION_NAME,
+                                                                asHexedAddress(tokenID.get()))
+                                                        .payingWith(ACCOUNT)
+                                                        .via(
+                                                                "unpauseFungibleAccountDoesNotOwnPauseKeyFailingTxn")
+                                                        .gas(GAS_TO_OFFER)
+                                                        .hasKnownStatus(CONTRACT_REVERT_EXECUTED),
+                                                cryptoUpdate(ACCOUNT).key(UNPAUSE_KEY),
+                                                contractCall(
+                                                                PAUSE_UNPAUSE_CONTRACT,
+                                                                UNPAUSE_TOKEN_ACCOUNT_FUNCTION_NAME,
+                                                                asHexedAddress(tokenID.get()))
+                                                        .payingWith(ACCOUNT)
+                                                        .via(UNPAUSE_FUNGIBLE_TXN)
+                                                        .gas(GAS_TO_OFFER))))
+                .then(
+                        childRecordsCheck(
+                                "unpauseFungibleAccountDoesNotOwnPauseKeyFailingTxn",
+                                CONTRACT_REVERT_EXECUTED,
+                                recordWith()
+                                        .status(INVALID_SIGNATURE)
+                                        .contractCallResult(
+                                                resultWith()
+                                                        .contractCallResult(
+                                                                htsPrecompileResult()
+                                                                        .withStatus(
+                                                                                INVALID_SIGNATURE)))),
+                        getTokenInfo(VANILLA_TOKEN).hasPauseStatus(Unpaused));
+    }
+
+    private HapiApiSpec pauseNonFungibleTokenHappyPath() {
+        final AtomicReference<TokenID> tokenID = new AtomicReference<>();
+
+        return defaultHapiSpec("PauseNonFungibleTokenHappyPath")
+                .given(
+                        newKeyNamed(PAUSE_KEY),
+                        cryptoCreate(TOKEN_TREASURY),
+                        cryptoCreate(ACCOUNT).balance(INITIAL_BALANCE),
+                        tokenCreate(VANILLA_TOKEN)
+                                .tokenType(NON_FUNGIBLE_UNIQUE)
+                                .treasury(TOKEN_TREASURY)
+                                .pauseKey(PAUSE_KEY)
+                                .adminKey(PAUSE_KEY)
```

Comment:
The naming here is a bit confusing. We have `PAUSE_KEY` set to two different types of keys. Probably you can name this key `MULTI_PURPOSE_KEY` or assign `ADMIN_KEY` as admin key.